pkgs/applications/audio/projectm/default.nix

```diff
@@ -24,7 +24,13 @@ stdenv.mkDerivation {
     sha256 = "1vh6jk68a0jdb6qwppb6f8cbgmhnv2ba3bcavzfd6sq06gq08cji";
   };
 
+  patch_gcc6 = fetchurl {
```

Comment:
Could you use `patches = [ ... ]`? It needs changing `patchPhase` -> `postPatch`

Also please use `fetchpatch`

Reply:
I can not use `patches` because this patch needs to be applied in `./src/libprojectM`, not in top level directory.

I have changed `fetchurl` to `fetchpatch`. Should we use `fetchpatch` for patches even when its `filterdiff` functionality is not necessary and `fetchurl` is fine?